src/NodaTime/Calendars/IslamicYearMonthDayCalculator.cs

```diff
@@ -41,22 +41,23 @@ internal sealed class IslamicYearMonthDayCalculator : RegularYearMonthDayCalcula
         /// <summary>The pattern of leap years within a cycle, one bit per year, for this calendar.</summary>
         private readonly int leapYearPatternBits;
 
-        private static readonly int[] TotalDaysByMonth;
+        private static readonly int[] TotalDaysByMonth = GenerateTotalDaysByMonth();
```

Comment:
again, a comment as to what this contains would be nice, since it's not immediately obvious whether the result is `[0, 29, ...]` or `[29, 59, ...]`.